cocos/core/event/callbacks-invoker.ts

```diff
@@ -276,13 +314,7 @@ export class CallbacksInvoker {
                 for (let i = 0; i < infos.length; ++i) {
                     const info = infos[i];
                     if (info && info.callback === callback && info.target === target) {
-                        if (list.isInvoking) {
-                            list.cancel(i);
-                        }
-                        else {
-                            fastRemoveAt(infos, i);
-                            callbackInfoPool.free(info);
-                        }
+                        list.cancel(i);
```

Comment:
Why not remove directly? Using cancel will much slower.

Reply:
If it's during invocation, we can't change the array, otherwise the for loop may skip some callbacks. The cancel is reimplemented with `isInvoking` check, will remove when it's possible

Reply:
cool